table/tables/partition.go

```diff
@@ -205,3 +205,52 @@ func (t *PartitionedTable) RemoveRecord(ctx sessionctx.Context, h int64, r []typ
 	tbl := t.GetPartition(pid)
 	return tbl.RemoveRecord(ctx, h, r)
 }
+
+// UpdateRecord implements table.Table UpdateRecord interface.
+// `touched` means which columns are really modified, used for secondary indices.
+// Length of `oldData` and `newData` equals to length of `t.WritableCols()`.
+func (t *PartitionedTable) UpdateRecord(ctx sessionctx.Context, h int64, currData, newData []types.Datum, touched []bool) error {
+	partitionInfo := t.meta.GetPartitionInfo()
+	old, err := t.locatePartition(ctx, partitionInfo, currData)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	new, err := t.locatePartition(ctx, partitionInfo, newData)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// The old and new data locate in different partitions.
+	// Remove record from old partition and add record to new partition.
+	if old != new {
+		_, err = t.GetPartition(new).AddRecord(ctx, newData, false)
```

Comment:
Should we set `StmtAddDirtyTableOP` here?

Reply:
No.
1. It's caller's responsibility to call `StmtAddDirtyTableOP`. `executor` package should do it, rather than `table` package.
2. I'll handle dirty table and binlog in other PR